### PR TITLE
created footer component | Footer Early Access

### LINKED ISSUE #21: Design footer

### PR DESCRIPTION
**Pull request by vonnerbosa**
## 🎟️ Issue
Fixes #21 
## 📝  Description
Basic layout for footer . Will update in the future

## 📷 Screenshots (if applicable)

image

image

## 📎 Checklist 

- [x] Footer
- [x] Payment Icons
- [x] Copyright
- [x] Dead Links

## 😝 Out of scope 
-Beautiful Footer

## 📢 Additional Information

I'm no longer 3-month footer man

### CODE REVIEW

File: frontend/components/shared/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+              <Link href="#" color="foreground" className="text-xs">
+                Service Guarantee
+              </Link>
+            </li>
+            <li className="list-none">
+              <Link href="#" color="foreground" className="text-xs">
+                Website Feedback
+              </Link>
+            </li>
+          </ul>
```
Comment (mendoza-allendg) [inline frontend/components/shared/Footer.tsx]: This block has been repeated multiple times. It's better to create a list of contactMethods first, then loop them like this:

contactMethods.map((contact) => (
            < li className="list-none">
              < Link href={contact.link} color="foreground" className="text-xs">
                {contact.Name}
              < /Link>
            < /li >
))

File: frontend/components/shared/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            <li className="list-none">
+              <Link href="#" color="foreground" className="text-xs">
+                Terms and Conditions
+              </Link>
+            </li>
+            <li className="list-none">
+              <Link href="#" color="foreground" className="text-xs">
+                Privacy and Cookies
+              </Link>
+            </li>
```
Comment (mendoza-allendg) [inline frontend/components/shared/Footer.tsx]: Same here

File: frontend/components/shared/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+              <Link href="#" color="foreground" className="text-xs">
+                Partner Portal
+              </Link>
+            </li>
+            <li className="list-none">
+              <Link href="#" color="foreground" className="text-xs">
+                Affiliate Program{' '}
+              </Link>
+            </li>
+          </ul>
```
Comment (mendoza-allendg) [inline frontend/components/shared/Footer.tsx]: Same here.

File: frontend/components/shared/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        </div>
+
+        <div className="flex flex-col p-4 text-sm mx-auto text-gray-500">
+          <div>Payment Methods</div>
+          <div className="flex flex-col flex-wrap justi">
+            <div className="flex justify-center">
+              <Image
```
Comment (mendoza-allendg) [inline frontend/components/shared/Footer.tsx]: Same here. Also, please put an alt prop in the image.

File: frontend/layouts/default.tsx [hunk 5]
```diff
@@ -27,16 +27,9 @@ export default function DefaultLayout({
...
-          href="https://nextui-docs-v2.vercel.app?utm_source=next-app-template"
-          title="nextui.org homepage"
-        >
-          <span className="text-default-600">Powered by</span>
-          <p className="text-primary">NextUI</p>
-        </Link>
+      <Footer />
+      <footer className="flex justify-center w-full text-slate-500 bg-gray-100 items-center text-sm p-2">
+        Copyright © 2024 DiveNGo.com All rights reserved
       </footer>
```
Comment (mendoza-allendg) [inline frontend/layouts/default.tsx]: You can add this inside the < Footer /> component

**Review by mendoza-allendg — CHANGES_REQUESTED** [review]
Nice PR! I've found some rooms for improvement, please check them! 🙇 

Overall, nice update!

Some bugs I found:
1. Footer dimensions are not aligned with header. Ultimately, we need to follow these dimensions in the app content with the exception of the SearchBar
Screenshot 2024-01-21 153823

2. The footer's responsiveness can be improved
Screenshot 2024-01-21 153916

3. The footer is fixed in its position. We want the footer to only show at the bottom of the page (after all contents have been scrolled through)

https://github.com/mendoza-allendg/sea-bit/assets/54106527/8deb6e44-f693-4c25-9cf9-bd09bded6202

**Comment by vonnerbosa** [comment]
Made improvements to the footer

File: frontend/components/shared/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
+import { Image, Link } from '@nextui-org/react'
+
+const contactMethods = [
```
Comment (mendoza-allendg) [inline frontend/components/shared/Footer.tsx]: [nits]

If a variable is fixed, and wont change dynamically in the application and is outside the component scope, its better to use all caps like:

CONTACT_METHODS. However, I think that we might need the CONTACT_METHODS variable in the future for more important stuff, so its better to use something like COMPANY_CONTACT_METHODS

File: frontend/components/shared/Footer.tsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    link: '#',
+  },
+  {
+    id: 2,
+    name: 'Website Feedback',
+    link: '#',
+  },
+]
+
+const companyInfo = [
```
Comment (mendoza-allendg) [inline frontend/components/shared/Footer.tsx]: [nits]

array variables should be plural

File: frontend/components/shared/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+export const Footer = () => {
+  return (
+    <div className="flex justify-center bg-gray-100 h-auto static bottom-0 border-t-1 border-solid border-gray-300 w-full">
+      <div className="flex flex-col justify-center items-center w-full">
+        <div className="flex-wrap container px-6 lg:px-20 xl:px-35 2xl:px-14 items-start flex justify-center w-full py-7">
+          <div className="flex flex-nowrap flex-1 justify-center text-xs mx-auto flex-shrink-0 max-w-md">
+            <div className="flex flex-col justify-start p-4">
+              <span className="font-bold text-base pb-1.5">Contact Us</span>
+              <ul>
+                {contactMethods.map((contact) => (
```
Comment (mendoza-allendg) [inline frontend/components/shared/Footer.tsx]: You don't have to manually put ids in the arrays. The .map() method provides the array's index and can be used as the key.

Try something like:
.map(contact, i) i is the index

**Review by mendoza-allendg — CHANGES_REQUESTED** [review]
Looking GOODDD!!
Just a few minor points then i think its all good to go! 💯

**Review by mendoza-allendg — APPROVED** [review]
LGTM!

**Pull request merged**